test/integration_test/migration_backup_test.go

```diff
@@ -0,0 +1,177 @@
+
+		// Change kubeconfig to destination cluster
+		err = dumpRemoteKubeConfig(remoteConfig)
+		require.NoErrorf(t, err, "Unable to write clusterconfig: %v", err)
+
+		err = setRemoteConfig(remoteFilePath)
+		require.NoError(t, err, "Error setting remote config")
+
+		// Backup app from the destination cluster to cloud
+		currBackupLocation, err := createBackupLocation(t, appKey+"-backup-location", testKey+"-mysql-migration", storkv1.BackupLocationType(location), secret)
+		require.NoError(t, err, "Error creating backuplocation %s", currBackupLocation.Name)
+
+		currBackup, err := createApplicationBackupWithAnnotation(t, "dest-backup", testKey+"-mysql-migration", currBackupLocation)
+		require.NoError(t, err, "Error creating app backups")
+
+		err = waitForAppBackupCompletion(currBackup.Name, currBackup.Namespace, applicationBackupSyncRetryTimeout)
+		require.NoError(t, err, "Application backup %s in namespace %s failed.", currBackup.Name, currBackup.Namespace)
+
+		destroyAndWait(t, []*scheduler.Context{preMigrationCtx})
+
+		logrus.Infoln("Completed backup of apps destination cluster")
+
+		// Switch kubeconfig to source cluster for restore
+		err = setRemoteConfig("")
+		require.NoError(t, err, "Error resetting remote config")
+
+		srcBackupLocation, err := createBackupLocation(t, appKey+"-backup-location", testKey+"-mysql-migration", storkv1.BackupLocationType(location), secret)
+		require.NoError(t, err, "Error creating backuplocation %s", currBackupLocation.Name)
+
+		// Set sync to true on first cluster so that backup from second cluster is available
+		srcBackupLocation.Location.Sync = true
+		srcBackupLocation, err = storkops.Instance().UpdateBackupLocation(srcBackupLocation)
+		require.NoError(t, err, "Failed to set backup-location sync to true")
+
+		// Check periodically to see if the backup from this test is synced on second cluster
+		var allAppBackups *storkv1.ApplicationBackupList
+		listBackupsTask := func() (interface{}, bool, error) {
+			allAppBackups, err = storkops.Instance().ListApplicationBackups(srcBackupLocation.Namespace)
+			if err != nil {
+				logrus.Infof("Failed to list app backups on first cluster post migrate and sync. Error: %v", err)
+				return "", true, fmt.Errorf("Failed to list app backups on first cluster")
+			} else if allAppBackups != nil && len(allAppBackups.Items) > 0 {
+				// backups sync has started, check if current backup has synced
+				backupToRestore := getBackupFromListWithAnnotations(allAppBackups, currBackup.Annotations[backupSyncAnnotation])
+				if backupToRestore != nil {
+					return "", false, nil
+				}
+			}
+			return "", true, fmt.Errorf("Failed to list app backups on first cluster")
+		}
+		_, err = task.DoRetryWithTimeout(listBackupsTask, applicationBackupSyncRetryTimeout, defaultWaitInterval)
+		require.NoError(t, err, "Error listing application backups")
+
+		backupToRestore := getBackupFromListWithAnnotations(allAppBackups, currBackup.Annotations[backupSyncAnnotation])
```

Comment:
These steps are similar to the ones being done in backupsync test, can we move it to a common function?

Reply:
Moved to a common method